lib/spack/spack/spec.py

```diff
@@ -1929,7 +1921,7 @@ def from_node_dict(node):
 
         # this spec may have been built with older packages than we have
         # on-hand, and we may not have the build dependencies, so mark it
-        # so we don't recompute full_hash and build_hash.
+        # so we don't recompute dag_hash.
```

Comment:
This was only necessary because there were specs that did not have `full_hash` or `build_hash`, and we didn't store package hashes or build deps.  It was meant to prevent recompilation, but now we don't need to do that -- there is *only* the DAG hash and it'll either be concrete or not.

So I think we can simplify and remove this.